cdc/puller/ddl_puller.go

```diff
@@ -42,6 +49,84 @@ const (
 	DDLPullerTableName = "DDL_PULLER"
 )
 
+// DDLJobPuller is used to pull ddl job from TiKV.
+type DDLJobPuller interface {
+	Puller
+
+	UnmarshalDDL(rawKV *model.RawKVEntry) (*timodel.Job, error)
+}
+
+type ddlJobPullerImpl struct {
+	Puller
+
+	tableInfo *model.TableInfo
+	columnID  int64
+}
+
+func (p *ddlJobPullerImpl) UnmarshalDDL(rawKV *model.RawKVEntry) (*timodel.Job, error) {
+	return entry.ParseJob(p.tableInfo, rawKV, p.columnID)
+}
+
+// NewDDLJobPuller create a new NewDDLJobPuller fetch event start from checkpointTs
+// and put into buf.
+func NewDDLJobPuller(
+	ctx context.Context,
+	pdCli pd.Client,
+	grpcPool kv.GrpcPool,
+	regionCache *tikv.RegionCache,
+	kvStorage tidbkv.Storage,
+	pdClock pdutil.Clock,
+	checkpointTs uint64,
+	cfg *config.KVClientConfig,
+	changefeed model.ChangeFeedID,
+) (DDLJobPuller, error) {
+	version, err := kvStorage.CurrentVersion(tidbkv.GlobalTxnScope)
+	if err != nil {
+		return nil, err
+	}
+	snap, err := kv.GetSnapshotMeta(kvStorage, version.Ver)
+	if err != nil {
+		return nil, err
+	}
+
+	dbInfos, err := snap.ListDatabases()
+	if err != nil {
+		return nil, cerror.WrapError(cerror.ErrMetaListDatabases, err)
+	}
+	spans := make([]regionspan.Span, 0, 3)
+	var tblInfo *model.TableInfo
+	var colID int64
+	for _, db := range dbInfos {
+		if db.Name.L == mysql.SystemDB {
+			tables, err := snap.ListTables(db.ID)
+			if err != nil {
+				return nil, cerror.WrapError(cerror.ErrMetaListDatabases, err)
+			}
+			for _, t := range tables {
+				if t.Name.L == "tidb_ddl_job" {
+					tblInfo = model.WrapTableInfo(db.ID, db.Name.L, 0, t)
+					spans = append(spans, regionspan.GetTableSpan(t.ID))
```

Comment:
Do we already make an assumption that `tidb` will be upgraded before `ticdc`?
I think we need some comments here in case that we make some midification in the future.